### PR TITLE
Remove setProps and replaceProps completely

### PR DESCRIPTION
**Pull request by koba04**
`setProps` and `replaceProps` are removed completely on v15.0

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
@koba04 updated the pull request.

File: src/renderers/shared/stack/reconciler/__tests__/ReactCompositeComponentState-test.js [hunk 1]
```diff
@@ -140,8 +140,7 @@ describe('ReactCompositeComponent-state', function() {
     );
     ReactDOM.render(
       <TestComponent stateListener={stateListener} nextColor="green" />,
-      container,
-      instance.peekAtCallback('setProps')
```
Comment (jimfb) [inline src/renderers/shared/stack/reconciler/__tests__/ReactCompositeComponentState-test.js]: I don't think we want to remove this one.  This isn't actually the same as the setProps that you removed elsewhere.

Comment (koba04) [inline src/renderers/shared/stack/reconciler/__tests__/ReactCompositeComponentState-test.js]: @jimfb Thank you! I got it fixed!

**Comment by facebook-github-bot-4** [comment]
@koba04 updated the pull request.

**Comment by jimfb** [comment]
Ok, this looks good, thanks @koba04!

**Comment by koba04** [comment]
Thank you!

**Comment by zpao** [comment]
Sorry to do this, but I'm going to revert this. Can you submit separate PRs for the changes to `docs/` and `src/` (this won't cherry-pick cleanly and will screw up some of the release tracking). In the future we should keep docs and code changes separate unless it's the initial introduction or removal of a feature (at least until we have a ship from master release process and we'll have other things to figure out then).

**Comment by koba04** [comment]
Thank you, I got it!
I'll submit separate PRs for this!

**Pull request merged**